src/main/java/tc/oc/pgm/events/PlayerTimedPunishmentEvent.java

```diff
@@ -0,0 +1,34 @@
+package tc.oc.pgm.events;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.time.Duration;
+import java.time.Instant;
+import org.bukkit.command.CommandSender;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.community.ModerationCommands.PunishmentType;
+
+/** Called when a punishment that expires after a duration of time is created * */
+public class PlayerTimedPunishmentEvent extends PlayerPunishmentEvent {
```

Comment:
Instead of having 2 events, remove this and add a Duration field to `PlayerPunishmentEvent`.
If the ban is permanent, then the Duration is MAX_VALUE.